btas/generic/linear_algebra.h

```diff
@@ -0,0 +1,281 @@
+//
+// Created by Karl Pierce on 1/26/20.
+//
+
+#ifndef BTAS_LINEAR_ALGEBRA_H
+#define BTAS_LINEAR_ALGEBRA_H
+#include <btas/error.h>
+
+#ifdef BTAS_HAS_CBLAS
+
+namespace btas{
+  /// Computes L of the LU decomposition of tensor \c A
+/// \param[in, out] A In: A reference matrix to be LU decomposed.  Out:
+/// The L of an LU decomposition of \c A.
+
+  template <typename Tensor> void LU_decomp(Tensor &A) {
+
+#ifndef BTAS_HAS_LAPACKE
+    BTAS_EXCEPTION("Using this function requires LAPACKE");
+#endif // BTAS_HAS_LAPACKE
+
+    if(A.rank() > 2){
+      BTAS_EXCEPTION("Tensor rank > 2. Can only invert matrices.");
+    }
+
+    btas::Tensor<int> piv(std::min(A.extent(0), A.extent(1)));
+    Tensor L(A.range());
+    Tensor P(A.extent(0), A.extent(0));
+    P.fill(0.0);
+    L.fill(0.0);
+
+    // LAPACKE LU decomposition gives back dense L and U to be
+    // restored into lower and upper triangular form, and a pivoting
+    // matrix for L
+    auto info = LAPACKE_dgetrf(LAPACK_ROW_MAJOR, A.extent(0), A.extent(1),
```

Comment:
This won't compile unless `BTAS_HAS_LAPACKE` is defined. The else clause of preprocessor if on line 18 needs to enclose the rest of the function body. Same pattern should apply everywhere else

Reply:
I believe I fixed all the instances of this issue